cosmos/plugin/airflow3.py

```diff
@@ -10,23 +10,34 @@
 from unittest.mock import patch
 from urllib.parse import urlsplit
 
+import airflow
 from airflow.configuration import conf
 from airflow.plugins_manager import AirflowPlugin
 from airflow.sdk import ObjectStoragePath
 from fastapi import FastAPI
 from fastapi.responses import HTMLResponse, JSONResponse, Response
+from packaging.version import Version
 
 from cosmos.constants import AIRFLOW_OBJECT_STORAGE_PATH_URL_SCHEMES
 from cosmos.plugin.snippets import IFRAME_SCRIPT
 
+# Airflow version gating: External views feature for the plugins used here (CosmosAF3Plugin) exist only in >= 3.1
+AIRFLOW_VERSION = Version(airflow.__version__)
+
+
+def ensure_airflow_version_supported() -> None:
+    if AIRFLOW_VERSION < Version("3.1.0"):
+        raise RuntimeError(
+            "Cosmos AF3 plugin requires Airflow >= 3.1. External views are unavailable on earlier versions."
+        )
+
+
 API_BASE = conf.get("api", "base_url", fallback="")  # reads AIRFLOW__API__BASE_URL
 API_BASE_PATH = urlsplit(API_BASE).path.rstrip("/")
 
 
-# TODO: This context manager and its usage in the method "_read_content_via_object_storage" has been added due
-# to the current limitation in Airflow 3.1.0 where plugins are not able to resolve connections via the API server.
-# Once this is fixed, potentially in PR https://github.com/apache/airflow/pull/56602 planned to be released in
-# Airflow 3.1.1, test the fix and remove this context manager and its usage.
+# Note: Airflow 3.1.0 and 3.1.1 had a limitation where plugins could not resolve connections via the API server.
```

Comment:
Thanks a lot, @pankajkoti , it's great that this will be solved in Airflow 3.1.2. Did you validate this with https://pypi.org/project/apache-airflow/3.1.2rc1/?